src/NuGet.Core/NuGet.Packaging/Signing/Native/CmsFactory.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography.Pkcs;
+
+namespace NuGet.Packaging.Signing
+{
+    internal static class CmsFactory
+    {
+        internal static ICms Create(byte[] cmsBytes)
+        {
+#if IS_SIGNING_SUPPORTED
+            ICms cms = null;
+#if IS_DESKTOP
+            NativeCms nativeCms = NativeCms.Decode(cmsBytes);
+            cms = new NativeCmsWrapper(nativeCms);
+#elif IS_CORECLR
```

Comment:
Since you added the outer `#if IS_SIGNING_SUPPORTED` is `IS_CORECLR` necessary here?  Can you just have `#else` like [here](https://github.com/NuGet/NuGet.Client/pull/3168/files#diff-2b8de836d50e305e85d33ce4c6d179f9R17-R35).

Reply:
Thanks! Changed accordingly.